### DISCUSSION TITLE
Is there an option to save `study` during the optimization process?

### DISCUSSION THREAD

**Discussion by DavidHarar** [optuna/optuna#4537] [Q&A] [answered]
### Don't use GitHub Issues to ask support questions.

Hello, 

First, thank you so much for this great package, it is amazing!

Is that by any chance an option to save the study while being trained? kind of saving its log, or only the best experiment so far.
All the examples I have encountered so far showed how to extract the best model parameters only after the tuning has ended.

Thank you,
David

**Comment by nzw0301** [comment, marked as answer]
Like this?

```python
import optuna
import joblib


class SaveStudyCallback:
    def __init__(self):
        pass

    def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -> None:
        i = trial.number
        joblib.dump(study, f"study_{i}.pkl")


def objective(trial):

    return trial.suggest_float("x", 0, 1)


study_stop_cb = SaveStudyCallback()
study = optuna.create_study()
study.optimize(objective, n_trials=10, callbacks=[study_stop_cb])

study = joblib.load("study_5.pkl")
assert len(study.get_trials()) == 6
```

**Reply by DavidHarar** [reply]
That works for me. Thanks so much!
